pkg/cri/resource-manager/policy/builtin/memtier/pools_test.go

```diff
@@ -33,23 +33,37 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
 )
 
-func findNodeWithID(id int, nodes []Node) *Node {
+func findNodeWithID(id int, nodes []Node) Node {
 	for _, node := range nodes {
 		if node.NodeID() == id {
-			return &node
+			return node
 		}
 	}
 	panic("No node found")
 }
 
 func setLinks(nodes []Node, tree map[int][]int) {
+	parents := map[int]int{}
```

Comment:
I got a bit confused by the naming and the fact that the parent id isn't used for anything. We actually want to have a list of children, (i.e. nodes that have a parent). Dunno if
```
nonOrphans := make(map[int]struct{})
```
would be more self-evident(?)

Anyway, just a suggestion. We can go with this implementation, too